### PR TITLE
Expose whenReady and extrinsics/storage

### PR DESCRIPTION
**Pull request by jacogr**
- Closes https://github.com/polkadot-js/api/issues/198
- Related #161

### CODE REVIEW

File: packages/api-observable/src/Base.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@ import { Method } from '@polkadot/jsonrpc/types';
 
 import { Observable, combineLatest } from 'rxjs';
 import { defaultIfEmpty, map } from 'rxjs/operators';
+import extrinsicsFromMeta from '@polkadot/extrinsics/fromMetadata';
+import extrinsicsStatic from '@polkadot/extrinsics/static';
+import storageFromMeta from '@polkadot/storage/fromMetadata';
+import storageStatic from '@polkadot/storage/static';
+import { Hash } from '@polkadot/types/index';
```
Comment (amaurymartiny) [inline packages/api-observable/src/Base.ts]: Not sure how strict you are about this, but I tend to sort imports alphabetically by import name.

File: packages/api-observable/src/Base.ts [hunk 2]
```diff
@@ -19,10 +24,43 @@ const defaultMapFn = (result: any): any =>
 // Raw base implementation for the observable API. It simply provides access to raw calls, allowing
 // decendants to make direct queries to either API methods or actual storage
 export default class ApiBase {
-  protected api: RxApiInterface;
+  protected _api: RxApiInterface;
+  protected _genesisHash: Hash;
+
+  // Promise that resolves the first time we are connected and loaded
+  whenReady: Promise<boolean>;
```
Comment (amaurymartiny) [inline packages/api-observable/src/Base.ts]: This package is api-observable, wouldn't it make more sense to return an Observable here?

Usage:
`api.whenReady().then(...)` -> `api.whenReady().subscribe(...)`, so not a big difference for devs

File: packages/api-observable/src/Base.ts (hunk 1 above)
Comment (jacogr) [reply packages/api-observable/src/Base.ts]: I am :( Mucked up there.

File: packages/api-observable/src/Base.ts (hunk 2 above)
Comment (jacogr) [reply packages/api-observable/src/Base.ts]: You are right.

File: packages/api-observable/src/Base.ts (hunk 1 above)
Comment (jacogr) [reply packages/api-observable/src/Base.ts]: I normally (try and) sort by "package imported from"

File: packages/api-observable/src/Queries.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { AccountId, Balance, Bool, BlockNumber, Index, Moment, Perbill, PropIndex, ReferendumIndex, u32 } from '@polkadot/types/index';
 import { Tuple } from '@polkadot/types/codec';
-import storage from '@polkadot/storage/testing';
+import storage from '@polkadot/storage/static';
```
Comment (amaurymartiny) [inline packages/api-observable/src/Queries.ts]: import order.

File: packages/api-provider/src/mock/mocks.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { MockState, MockState$Db, MockState$Subscriptions } from './types';
 
 import BN from 'bn.js';
 import Header from '@polkadot/types/Header';
-import storage from '@polkadot/storage/testing';
+import storage from '@polkadot/storage/static';
```
Comment (amaurymartiny) [inline packages/api-provider/src/mock/mocks.ts]: import order

File: packages/api-rx/test/e2e/subscribe.test.js [hunk 5]
```diff
@@ -2,7 +2,7 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
-import storage from '@polkadot/storage/testing';
+import storage from '@polkadot/storage/static';
```
Comment (amaurymartiny) [inline packages/api-rx/test/e2e/subscribe.test.js]: import order

File: packages/api/test/e2e/state.test.js [hunk 6]
```diff
@@ -2,7 +2,7 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
-import storage from '@polkadot/storage/testing';
+import storage from '@polkadot/storage/static';
```
Comment (amaurymartiny) [inline packages/api/test/e2e/state.test.js]: import order

File: packages/api/test/e2e/subscribe.test.js [hunk 7]
```diff
@@ -2,7 +2,7 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
-import storage from '@polkadot/storage/testing';
+import storage from '@polkadot/storage/static';
```
Comment (amaurymartiny) [inline packages/api/test/e2e/subscribe.test.js]: import order

**Pull request merged**